### PR TITLE
Improve application endpoint

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Revolyssup** [comment]
@piyushsingariya  , @Abhishek-kumar09 is changing the way applications are handled at the UI and this PR gets rid of some of the issues as well as introduces this new field "format". We can discuss the changes on a call. Although the new behaviour in the UI will be needed across mesheryctl as well
// @navendu-pottekkat

**Comment by Revolyssup** [comment]
This will also require changes in migration files(table structure) for application table

File: handlers/meshery_application_handler.go [hunk 1]
```diff
@@ -109,18 +109,6 @@ func (h *Handler) handleApplicationPOST(
...
-			obj := "save"
-			h.log.Error(ErrApplicationFailure(err, obj))
-			http.Error(rw, ErrApplicationFailure(err, obj).Error(), http.StatusBadRequest)
-			return
-		}
-
-		// Assign a name if no name is provided
-		if parsedBody.ApplicationData.Name == "" {
-			parsedBody.ApplicationData.Name = applicationName
-		}
```
Comment (Abhishek-kumar09) [inline handlers/meshery_application_handler.go:123]: We have to make sure as the client that we are passing name field. @piyushsingariya 
like: 
```js
{
	"application_data": {
		"name": "Any name",
		"application_file": ".....",
		"format": "pattern" // default is pattern
	}
}
```

**Review by Abhishek-kumar09 — APPROVED** [review]
Thanks for the PR @Revolyssup

File: models/default_local_provider.go [hunk 2]
```diff
@@ -1076,13 +1076,9 @@ func githubRepoApplicationScan(
 		RegisterFileInterceptor(func(f walker.File) error {
 			ext := filepath.Ext(f.Name)
 			if ext == ".yml" || ext == ".yaml" {
-				name, err := GetApplicationName(string(f.Content))
-				if err != nil {
-					return err
-				}
```
Comment (leecalcote) [inline models/default_local_provider.go:1079]:
```suggestion
```

File: handlers/meshery_application_handler.go (hunk 1 above)
Comment (Revolyssup) [reply handlers/meshery_application_handler.go:123]: Yes that is right as we can't extract name from K8s manifest

**Pull request merged**